src/app/leaves/add/components/LeaveRequestFormBuilder.tsx

```diff
@@ -0,0 +1,271 @@
+"use client";
+import React from "react";
+import FormWrapper from "@/common/FormWrapper";
+import { cn } from "@/lib/utils";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { CalendarIcon } from "@radix-ui/react-icons";
+import { useForm } from "react-hook-form";
+import { z } from "zod";
+import { format } from "date-fns";
+import { Button } from "@/components/ui/button";
+import { Calendar } from "@/components/ui/calendar";
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form";
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "@/components/ui/popover";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+import { Textarea } from "@/components/ui/textarea";
+import { MembersProps } from "@/types";
+import { LeaveTypes } from "@/constant/constant";
+import { MemberSelector } from "@/common/MemberSelector/MemberSelector";
+
+export const leaveRequestSchema = z
+  .object({
+    memberId: z.string().min(1, {
+      message: "Member must be selected.",
+    }),
+    leaveType: z.string().min(1, {
+      message: "Leave Type must be selected.",
+    }),
+    startDate: z
+      .date({
+        message: "Start Date must be selected.",
+      })
+      .refine((date) => date >= new Date(), {
+        message:
+          "Start date must be greater than or equal to the current date.",
+      }),
+    endDate: z
+      .date({
+        message: "End Date must be selected.",
+      })
+      .refine((date) => date >= new Date(), {
+        message: "End date must be greater than or equal to the current date.",
+      }),
+    reasonForLeave: z.string().min(10, {
+      message: "Please provide a reason for leave.",
+    }),
+    status: z.string().min(1, {
+      message: "Status must be selected.",
+    }),
+  })
+  .refine((data) => data.endDate > data.startDate, {
+    message: "End date cannot be earlier than start date.",
+    path: ["endDate"],
+  });
+
+const leaveRequestType = [
+  {
+    leaveTypes: LeaveTypes.Personal,
+  },
+  {
+    leaveTypes: LeaveTypes.Sick,
+  },
+  {
+    leaveTypes: LeaveTypes.Vacation,
+  },
+];
+
+export const LeaveRequestFormBuilder = ({
+  members,
+}: {
+  members: MembersProps;
+}) => {
+  const form = useForm<z.infer<typeof leaveRequestSchema>>({
+    resolver: zodResolver(leaveRequestSchema),
+    defaultValues: {
+      memberId: "",
+      leaveType: "",
+      startDate: undefined,
+      endDate: undefined,
+      reasonForLeave: "",
+      status: "Pending",
+    },
+    mode: "all",
+  });
+
+  const { errors, isValid } = form.formState;
+
+  function onSubmit(values: z.infer<typeof leaveRequestSchema>) {
+    console.log(values);
+  }
+  return (
+    <FormWrapper>
+      <h1 className="text-sm text-sm font-semibold text-gray-800 my-4">
```

Comment:
Do you think we can create a separate component called `Title` for this purpose? I think we can use it in a lot of places.